src/cpp/rtps/security/logging/BuiltinLogging.cpp

```diff
@@ -0,0 +1,66 @@
+#include "fastdds/rtps/security/logging/BuiltinLogging.h"
+
+namespace eprosima {
+namespace fastrtps {
+namespace rtps {
+namespace security {
+
+BuiltinLogging::BuiltinLogging()
+  : stop_(false)
+  , thread_([this]() {
+      BuiltinLoggingType msg;
+      for (;;)
+      {
+        // Put the thread asleep until there is
+        // something to process
+        MessagePtr p = queue_.wait_pop();
+
+        if (!p)
+        {
+          if (stop_)
```

Comment:
If you follow my advice about the queue's destructor and notification, you should be able to do away with this member variable entirely and just use the invalid return value as the signal that the thread should shutdown.